docs/_ext/custom_styles/utils.py

```diff
@@ -70,54 +70,59 @@ def _parse_option_field(
             target_params_description.append(line)
 
     # find missing parameters
-    missing = set(target_args) - described_params
+    missing = target_args - described_params
 
-    return target_params_description, list(missing)
+    return target_params_description, missing
 
 
 def _generate_options_documentation(
-    current_class: object,
+    current_class: Type,
     method_name: str,
     target_args: List[str] = None,
     config: SphinxConfig = None,
     indent: str = "",
 ) -> List[str]:
     """Automatically generate documentation from the default options method."""
-
-    if current_class == object:
-        # check if no more base class
-        raise Exception(f"Option docstring for {', '.join(target_args)} is missing.")
-
     options_docstring_lines = []
 
-    default_opts = getattr(current_class, method_name, None)
-    if not default_opts:
-        # getter option is not defined
-        return []
-
     if not target_args:
-        target_args = list(default_opts().__dict__.keys())
-
-    # parse default options method
-    parsed_lines, target_args = _parse_option_field(
-        docstring=default_opts.__doc__ or "",
-        config=config,
-        target_args=target_args,
-        indent=indent,
-    )
-
-    if target_args:
-        # parse parent class method docstring if some arg documentation is missing
-        parent_parsed_lines = _generate_options_documentation(
-            current_class=inspect.getmro(current_class)[1],
-            method_name=method_name,
-            target_args=target_args,
+        default_opts_clsmethod = getattr(current_class, method_name, None)
+        if not default_opts_clsmethod:
+            # getter option is not defined
+            return []
+        target_args = set(default_opts_clsmethod().__dict__.keys())
+
+    mro_classes = inspect.getmro(current_class)
+    for i, mro_cls in enumerate(mro_classes):
+        default_opts_clsmethod = getattr(mro_cls, method_name, None)
+        if not default_opts_clsmethod:
+            continue
+        parsed_lines, target_args = _parse_option_field(
+            docstring=default_opts_clsmethod.__doc__ or "",
             config=config,
+            target_args=target_args,
             indent=indent,
         )
-        options_docstring_lines.extend(parent_parsed_lines)
-
-    options_docstring_lines.extend(parsed_lines)
+        if parsed_lines:
+            if i == 0:
+                description = "defined in the current class"
+            else:
+                description = "inherited from the parent class"
+            options_docstring_lines.extend(
+                [
+                    f"(Options {description} :class:`.{mro_cls.__name__}`)",
```

Comment:
I was also thinking of this. However, the structure of sections, i.e. `Experiment Options, Transpiler Options, Run Options`, became difficult to understand because these section tiles (`.. rubric::`) and added plain text are the same font size. Do you know better Sphinx directive for these section titles?

Reply:
Hmm, what about changing the `.. rubric::` section titles to headings, and changing this line to `.. rubric::`? Then it would look something like this:
![image](https://user-images.githubusercontent.com/3870315/225455546-1d73dee3-3c2c-4fed-a290-966c2b59a250.png)
I think it's nice to promote the section titles to headings so they can be linked to easily, and in theory the user can also quickly jump to a section using the sidebar.

Also, I wonder if we should move the default option value into the option description and remove the dropdown, would that look cleaner? Something like,

- **max_circuits** (_Optional[int]_, default: `None`) – The maximum number of circuits per job when running an experiment on a backend.

Reply:
That's good idea. Default value can be sometime very long (e.g. ndarray of length 100 to represent scanned value), sometime without pretty printing, and such value make documentation unnecessary long.

Reply:
I just remembered I tried section heading before. The underline of section title breaks build in section of parameters (i.e. `Args:`) because we are likely mixing up the [Google style and Numpy style](https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html#google-vs-numpy). As you posted, Parameter formatting is now broken and they are shown like `type lengths:` in your screen shot.

Reply:
I don't know how we can use sphinx general section and napoleon built-in section in the same docstring.

Reply:
Regarding the default value
![image](https://user-images.githubusercontent.com/39517270/225779356-599fcd7e-efe3-40e7-9e78-935b55a83bd4.png)
how do we want to show a value like this?

Reply:
Hmm, I guess there's no easy fix then. We can leave it as possible future work.

For default values, I think showing `numpy.linspace(-0.95, 0.95, 51)` is a lot more readable than that block. But is there a way to do that besides manually calculating the start end and step size from the array? Or we can add an optional field in the experiment options docstring allowing override of the default display value in cases like this?

Reply:
It's likely tough to get args for ndarray factory. I implemented truncation in [b1936a3](https://github.com/Qiskit/qiskit-experiments/pull/1077/commits/b1936a3d3cf9716e92c667ca484c9cc3828dc1ab)